sql/src/main/java/io/crate/metadata/functions/Signature.java

```diff
@@ -231,6 +237,29 @@ private Signature(FunctionName name,
         this.allowCoercion = allowCoercion;
     }
 
+    public Signature(StreamInput in) throws IOException {
+        name = new FunctionName(in);
+        kind = FunctionInfo.Type.values()[in.readVInt()];
+        int argsSize = in.readVInt();
+        argumentTypes = new ArrayList<>(argsSize);
+        for (int i = 0; i < argsSize; i++) {
+            argumentTypes.add(new TypeSignature(in));
```

Comment:
Is this safe to stream `TypeSignature` like this? I think this could lose information if it's  `ObjectParameterTypeSignature`

Reply:
👍 ah, good catch